spring-app/src/main/kotlin/pl/starchasers/up/security/JwtTokenFilter.kt

```diff
@@ -5,26 +5,46 @@ import org.springframework.security.core.GrantedAuthority
 import org.springframework.security.core.authority.SimpleGrantedAuthority
 import org.springframework.security.core.context.SecurityContextHolder
 import org.springframework.web.filter.GenericFilterBean
+import org.springframework.web.util.WebUtils
 import pl.starchasers.up.service.JwtTokenService
-import pl.starchasers.up.service.ROLE_KEY
+import pl.starchasers.up.util.SetCookieHeaderValueBuilder
 import javax.servlet.FilterChain
 import javax.servlet.ServletRequest
 import javax.servlet.ServletResponse
 import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
 
 class JwtTokenFilter(
     private val tokenService: JwtTokenService
 ) : GenericFilterBean() {
 
     override fun doFilter(request: ServletRequest?, response: ServletResponse?, chain: FilterChain) {
```

Comment:
🤷

Reply:
This has to stay like this:
- If someone has invalid tokens he won't be able to do anything anyway,
- There is an edge case where if user tries to log in with invalid tokens still present we clear the invalid tokens and in the same transaction set new tokens. This might lead to unpredictable results